clients/src/test/java/org/apache/kafka/clients/consumer/internals/CommitRequestManagerTest.java

```diff
@@ -226,25 +238,115 @@ public void testOffsetFetchRequest_EnsureDuplicatedRequestSucceed() {
+            new OffsetAndMetadata(0));
+
+        CommitRequestManager.OffsetCommitRequestState requestState = sendAndVerifyOffsetCommitRequests(
+            commitRequestManger,
+            offsets,
+            error);
+
+        assertExceptionHandling(commitRequestManger, requestState, error);
+        assertCoordinatorDisconnect(error);
+    }
+
+    @Test
+    public void testEnsureBackoffRetryOnOffsetCommitRequestTimeout() {
+        CommitRequestManager commitRequestManger = create(true, 100);
+        when(coordinatorRequestManager.coordinator()).thenReturn(Optional.of(mockedNode));
+
+        Map<TopicPartition, OffsetAndMetadata> offsets = Collections.singletonMap(new TopicPartition("topic", 1),
+            new OffsetAndMetadata(0));
+
+        commitRequestManger.addOffsetCommitRequest(offsets);
+        NetworkClientDelegate.PollResult res = commitRequestManger.poll(time.milliseconds());
+        assertEquals(1, res.unsentRequests.size());
+        res.unsentRequests.get(0).handler().onFailure(time.milliseconds(), new TimeoutException());
+
+        assertTrue(commitRequestManger.pendingRequests.hasUnsentRequests());
+        assertEquals(1, commitRequestManger.unsentOffsetCommitRequests().size());
+        long retryBackoffMs = commitRequestManger.unsentOffsetCommitRequests().peek().remainingBackoffMs(time.milliseconds());
+        assertRetryBackOff(commitRequestManger, retryBackoffMs);
+    }
+
+    private void assertCoordinatorDisconnect(final Errors error) {
+        if (error.exception() instanceof DisconnectException) {
+            verify(coordinatorRequestManager).markCoordinatorUnknown(any(), any());
+        }
+    }
+
+    private void assertExceptionHandling(CommitRequestManager commitRequestManger, CommitRequestManager.OffsetCommitRequestState requestState, Errors errors) {
+        long remainBackoffMs = requestState.remainingBackoffMs(time.milliseconds());
+        switch (errors) {
+            case NOT_COORDINATOR:
+            case COORDINATOR_NOT_AVAILABLE:
+            case REQUEST_TIMED_OUT:
+                verify(coordinatorRequestManager).markCoordinatorUnknown(any(), anyLong());
+                assertPollDoesNotReturn(commitRequestManger, remainBackoffMs);
+                break;
+            case UNKNOWN_TOPIC_OR_PARTITION:
+            case COORDINATOR_LOAD_IN_PROGRESS:
+                assertRetryBackOff(commitRequestManger, remainBackoffMs);
+                break;
+            case GROUP_AUTHORIZATION_FAILED:
+                // failed
+                break;
+            case TOPIC_AUTHORIZATION_FAILED:
+            case OFFSET_METADATA_TOO_LARGE:
+            case INVALID_COMMIT_OFFSET_SIZE:
+                // fail
+                // TODO: maybe we should retry
+                assertPollDoesNotReturn(commitRequestManger, Long.MAX_VALUE);
+                break;
+            case FENCED_INSTANCE_ID:
+                // what should we do
```

Comment:
Maybe make this a TODO if it's still unclear what to do. But then that should go into the main source code, not the test, right? Similar above.